gui/widgets/editable.cpp

```diff
@@ -63,15 +63,21 @@ void EditableWidget::reflowLayout() {
 }
 
 void EditableWidget::setEditString(const String &str) {
-	// TODO: We probably should filter the input string here,
-	// e.g. using tryInsertChar.
-	_editString = str;
+	_editString = "";
+	int pos = 0;
+	for (unsigned int i = 0; i < str.size(); i++)
+		if (tryInsertChar((byte)str[i], pos))
```

Comment:
inserting string char by char like this is extremely ineffective. Please make it in a single call.